Remotion/Data/DomainObjects.UnitTests/Mapping/TypeDefinitionHierarchyTest.cs

```diff
@@ -0,0 +1,105 @@
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Mapping;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Mapping
+{
+  [TestFixture]
+  public class TypeDefinitionHierarchyTest
+  {
+    [Test]
+    public void ArePartOfSameHierarchy ()
+    {
+      var iOther = InterfaceDefinitionObjectMother.CreateInterfaceDefinition();
+      var iExtendedOther = InterfaceDefinitionObjectMother.CreateInterfaceDefinition(extendedInterfaces: new[] { iOther });
+      var iDomainBase = InterfaceDefinitionObjectMother.CreateInterfaceDefinition();
+      var domainBase = ClassDefinitionObjectMother.CreateClassDefinition("Base", implementedInterfaces: new[] { iDomainBase });
+      var person = ClassDefinitionObjectMother.CreateClassDefinition("Person", baseClass: domainBase);
+      var customer = ClassDefinitionObjectMother.CreateClassDefinition("Customer", baseClass: person, implementedInterfaces: new[] { iExtendedOther });
+      var employee = ClassDefinitionObjectMother.CreateClassDefinition("Employee", baseClass: person);
+      var iCompany = InterfaceDefinitionObjectMother.CreateInterfaceDefinition(extendedInterfaces: new[] { iDomainBase });
+
+      var separate = ClassDefinitionObjectMother.CreateClassDefinition("Separate");
+
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(separate, iOther), Is.False);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(iOther, separate), Is.False);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(iOther, person), Is.False);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(person, iOther), Is.False);
+
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(customer, employee), Is.True);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(employee, customer), Is.True);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(employee, iCompany), Is.True);
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(iCompany, employee), Is.True);
+    }
+
+    [Test]
+    public void ArePartOfSameHierarchy_WithSameTypeDefinition_ReturnsTrue ()
+    {
+      var typeDefinition = TypeDefinitionObjectMother.CreateInterfaceDefinition();
+
+      Assert.That(TypeDefinitionHierarchy.ArePartOfSameHierarchy(typeDefinition, typeDefinition), Is.True);
+    }
+
+    [Test]
+    public void GetHierarchyRoots ()
```

Comment:
Dropping GetHierarchyRoots for TypeDefinitions in favor of only doing ClassDefinitions is in a later feature branch, isn't it?